### PR TITLE
llmproxy: check token size before auth

### PR DESCRIPTION
**Pull request by evict**
Before we pass the token along, we enforce a size check to make sure we don't cache any bogus. 

## Test plan
I've updated the unittests and all checks passed.

### CODE REVIEW

File: enterprise/cmd/llm-proxy/internal/actor/productsubscription/productsubscription.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ var (
 	minUpdateInterval = 10 * time.Minute
 
 	defaultUpdateInterval = 24 * time.Hour
+	tokenLength           = 68
```
Comment (bobheadxi) [inline enterprise/cmd/llm-proxy/internal/actor/productsubscription/productsubscription.go:24]:
```suggestion
	// product subscription tokens are always a prefix of 4 characters (sgs_)
	// followed by a 64-character hex-encoded SHA256 hash
	tokenLength = 4 + 64
```

File: enterprise/cmd/llm-proxy/internal/actor/productsubscription/productsubscription.go [hunk 2]
```diff
@@ -21,6 +21,9 @@ var (
 	minUpdateInterval = 10 * time.Minute
 
 	defaultUpdateInterval = 24 * time.Hour
+	// product subscription tokens are always a prefix of 4 characters (sgs_)
+	// followed by a 64-character hex-encoded SHA256 hash
+	tokenLength = 4 + 64
```
Comment (unknwon) [inline enterprise/cmd/llm-proxy/internal/actor/productsubscription/productsubscription.go:26]: nit: we wanna this actually be a `const` not `var` right? 😁

**Review by unknwon — APPROVED** [review]
Classic security PR 😁

File: enterprise/cmd/llm-proxy/internal/actor/productsubscription/productsubscription.go [hunk 3]
```diff
@@ -49,7 +52,7 @@ func (s *Source) Name() string { return llmproxy.ProductSubscriptionActorSourceN
 
 func (s *Source) Get(ctx context.Context, token string) (*actor.Actor, error) {
 	// "sgs_" is productSubscriptionAccessTokenPrefix
-	if token == "" || !strings.HasPrefix(token, "sgs_") {
+	if token == "" || !strings.HasPrefix(token, "sgs_") || len(token) != tokenLength {
 		return nil, actor.ErrNotFromSource{}
 	}
```
Comment (bobheadxi) [inline enterprise/cmd/llm-proxy/internal/actor/productsubscription/productsubscription.go:57]: actually - should this be part of the initial check? This returns an error indicating that this source does not care about this token, but this token has already claimed to be a relevant token due to its prefix. maybe we should raise a separate error:

```suggestion
	if token == "" || !strings.HasPrefix(token, "sgs_") {
		return nil, actor.ErrNotFromSource{}
	}
	
	if len(token) != tokenLength {
		return nil, errors.New("invalid token format")
	}
```

Comment (evict) [reply enterprise/cmd/llm-proxy/internal/actor/productsubscription/productsubscription.go:57]: You are right, that does make sense to me. I'll commit your suggestion.

File: enterprise/cmd/llm-proxy/internal/actor/productsubscription/productsubscription.go [hunk 4]
```diff
@@ -17,6 +17,8 @@ import (
 	"github.com/sourcegraph/sourcegraph/lib/errors"
 )
 
+const tokenLength = 68
```
[the remark this answers is not in the document]
Comment (evict) [reply enterprise/cmd/llm-proxy/internal/actor/productsubscription/productsubscription.go:20]: woops, just noticed. Will resolve.

**Review by bobheadxi — APPROVED** [review]
nice, thank you!

**Pull request merged**